code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -378,6 +380,45 @@
 	anchored = FALSE
 	target = null
 
+
+/mob/living/simple_animal/bot/cleanbot/larry/attackby(obj/item/I, mob/living/user)
+	if(I) //Does the arg exist?
+		if(istype(I, /obj/item/kitchen/knife)) //Is it a knife?
+			if(!knife)
+				var/obj/item/kitchen/knife/newknife = I
+				knife = newknife
+				newknife.forceMove(src)
+				message_admins("[user] attached a [newknife.name] to [src]") //This should definitely be a notified thing.
+				src.AddComponent(/datum/component/larryknife, knife.force)
+				update_icons()
+	else
+		return ..()
```

Comment:
Pretty sure this makes larry unattackable with melee weapons, since the ..() call does the actual attack, meaning if attacked with anything it will not return ..().
Additionally istype(...) checks for nulls, so there is no need to do if(I && istype(I)) since it simplifies to if(istype(I)).
Also you should probably log somewhere a knife was added to larry and add a to_chat message if it already has one.

Reply:
Addressed by careful formatting of if statement